src/oic/oauth2/util.py

```diff
@@ -57,7 +57,10 @@ def get_or_post(uri, method, req, content_type=DEFAULT_POST_CONTENT_TYPE,
     if method in ["GET", "DELETE"]:
         _qp = req.to_urlencoded()
         if _qp:
-            path = uri + '?' + _qp
+            if '?' in uri:
```

Comment:
I think it might be safer to use `urlsplit` and then append to the `query` part of the result and the use `urlunsplit`.

Reply:
You can't append to the query part of a Splitresult but I've tried to something similar.

Reply:
Yes, that is true. This works, but it can be simplified even more.

If you move the `urlunsplit` from the condition you can then keep the update part and in the `else` branch simpli assign the `_qs` to `_query`.

Reply:
I get your point and I think I actually dealt with in the latest version of the PR.

Reply:
Actually not, I have commented on the appropriate line.